scripts/clang-tidy.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -eu
+set -o pipefail
+
+# https://clang.llvm.org/extra/clang-tidy/
+
+: '
+Runs clang-tidy on the code in src/
+Return `1` if there are files automatically fixed by clang-tidy.
+Returns `0` if no fixes by clang-tidy.
+TODO: should also return non-zero if clang-tidy emits warnings
+or errors about things it cannot automatically fix. However I cannot
+figure out how to get this working yet as it seems that clang-tidy
+always returns 0 even on errors.
+'
+
+PATH_TO_CLANG_TIDY_SCRIPT="$(pwd)/mason_packages/.link/share/run-clang-tidy.py"
```

Comment:
Same; also what is `run-clang-tidy.py`?

Reply:
https://github.com/llvm-mirror/clang-tools-extra/blob/master/clang-tidy/tool/run-clang-tidy.py

It's bundled with `clang-tidy`, and does 1) parallel scanning of files, and 2) reads files to check from `compile_commands.json`